### PR TITLE
feat(@desktop/settings): Added advanced token settings

### LINKED ISSUE #13118: `BACKEND`: Assets `advance settings` backend (related to task #12611)
Related task #12611

Implement backend for advanced settings options and handle sync

### PR DESCRIPTION
**Pull request by Cuteivist**
Closes #13118 

Status-go change:  https://github.com/status-im/status-go/pull/4524

### What does the PR do

Added settings for advanced token settings based on https://github.com/status-im/status-desktop/issues/12611

### Affected areas

Settings

### CODE REVIEW

File: src/app_service/service/settings/service.nim [hunk 1]
```diff
@@ -508,6 +508,35 @@ QtObject:
...
+    return self.settings.displayAssetsBelowBalance
+
+  proc toggleDisplayAssetsBelowBalance*(self: Service): bool =
+    let newValue = not self.settings.displayAssetsBelowBalance
+    if(self.saveSetting(KEY_DISPLAY_ASSETS_BELOW_BALANCE, newValue)):
+      self.settings.displayAssetsBelowBalance = newValue
+      return true
+    return false
+
+  proc displayAssetsBelowBalanceThreshold*(self: Service): int64 =
```
Comment (caybro) [inline src/app_service/service/settings/service.nim:531]: Shouldn't this (and the respective QtProperty) be a float?

image

File: src/app/modules/main/wallet_section/all_tokens/view.nim [hunk 2]
```diff
@@ -160,4 +160,50 @@ QtObject:
...
+  proc getDisplayAssetsBelowBalanceThreshold(self: View): QVariant {.slot.} =
+    return newQVariant(self.delegate.getDisplayAssetsBelowBalanceThreshold())
+
+  proc setDisplayAssetsBelowBalanceThreshold(self: View, threshold: QVariant) {.slot.} =
+    self.delegate.setDisplayAssetsBelowBalanceThreshold(threshold.int64Val())
+    self.displayAssetsBelowBalanceThresholdChanged()
+
+  QtProperty[QVariant] displayAssetsBelowBalanceThreshold:
+    read = getDisplayAssetsBelowBalanceThreshold
+    write = setDisplayAssetsBelowBalanceThreshold
```
Comment (saledjenic) [inline src/app/modules/main/wallet_section/all_tokens/view.nim:202]: Do we really want this property to have a write accessor?

**Review by saledjenic — COMMENTED** [review]
@Cuteivist basically ok, left some minor comments, but wondering why we don't have a qml part for this?

File: src/app/modules/main/wallet_section/all_tokens/view.nim [hunk 3]
```diff
@@ -158,6 +158,53 @@ QtObject:
...
+      return
+    self.displayAssetsBelowBalanceChanged()
+
+  proc displayAssetsBelowBalanceThresholdChanged*(self: View) {.signal.}
+
+  proc getDisplayAssetsBelowBalanceThreshold(self: View): QVariant {.slot.} =
+    return newQVariant(self.delegate.getDisplayAssetsBelowBalanceThreshold())
+
+  proc setDisplayAssetsBelowBalanceThreshold(self: View, threshold: QVariant) {.slot.} =
+    if not self.delegate.setDisplayAssetsBelowBalanceThreshold(threshold.int64Val()):
```
Comment (caybro) [inline src/app/modules/main/wallet_section/all_tokens/view.nim:203]: It still expects an `int64` here?

Comment (caybro) [reply src/app/modules/main/wallet_section/all_tokens/view.nim:203]: Alright... the point is to use a `double` property, both in QML and in NIM; not sure if the latter is possible though :/ A regular `float` won't have the required precision for sure

**Review by Khushboo-dev-cpp — APPROVED** [review]
looks good!

File: src/app_service/service/settings/service.nim (hunk 1 above)
Comment (caybro) [reply src/app_service/service/settings/service.nim:531]: But the currency is coming from another user setting, not from status-go

File: src/app/modules/main/wallet_section/all_tokens/view.nim [hunk 4]
```diff
@@ -160,4 +160,50 @@ QtObject:
...
+    self.delegate.setDisplayAssetsBelowBalanceThreshold(threshold.int64Val())
+    self.displayAssetsBelowBalanceThresholdChanged()
+
+  QtProperty[QVariant] displayAssetsBelowBalanceThreshold:
+    read = getDisplayAssetsBelowBalanceThreshold
+    write = setDisplayAssetsBelowBalanceThreshold
+    notify = displayAssetsBelowBalanceThresholdChanged
+
+  proc displayAssetsBelowBalanceThresholdDecimals(self: View): int {.slot.} =
+    return 9
```
[the remark this answers is not in the document]
Comment (caybro) [reply src/app/modules/main/wallet_section/all_tokens/view.nim]: Shouldn't this be `2` by default, as it's in Figma?

**Pull request merged**